### PR TITLE
make description parameter of Guild.create_sticker non-Optional

### LINKED ISSUE #7311: Bad Request when creating a guild sticker.
### Summary

Getting an error when creating guild sticker.

### Reproduction Steps

See code below.

### Minimal Reproducible Code

```python
emoji2 = self.client.stickers[0] #Index 0 is an example here, but it happens with every index.
            async with aiohttp.ClientSession() as session:
                async with session.get(f"https://media.discordapp.net/stickers/{emoji2.id}.png") as resp:
                    file = discord.File(fp=BytesIO(await resp.read()), filename="sticker.png") 
            await ctx.guild.create_sticker(name=emoji2.name, emoji=emoji2.emoji, file=file, reason="FightMan01 bot matrica másolás")
```

### Expected Results

The sticker will be created.

### Actual Results

I'm getting an error: 
```
Traceback (most recent call last):
  File "/home/fightman/fightman01bot/mod.py", line 2808, in fork_sticker
    await ctx.guild.create_sticker(name=emoji2.name, emoji=emoji2.emoji, file=file, reason="FightMan01 bot matrica másolás")
  File "/home/fightman/.local/lib/python3.8/site-packages/discord/guild.py", line 2165, in create_sticker
    data = await self._state.http.create_guild_sticker(self.id, payload, file, reason)
  File "/home/fightman/.local/lib/python3.8/site-packages/discord/http.py", line 338, in request
    raise HTTPException(response, data)
discord.errors.HTTPException: 400 Bad Request (error code: 0): The browser (or proxy) sent a request that this server could not understand.
Traceback (most recent call last):
  File "/home/fightman/fightman01bot/mod.py", line 2809, in fork_sticker
    await ctx.guild.create_sticker(name=emoji2.name, emoji=emoji2.emoji, file=file, reason="FightMan01 bot matrica másolás")
  File "/home/fightman/.local/lib/python3.8/site-packages/discord/guild.py", line 2165, in create_sticker
    data = await self._state.http.create_guild_sticker(self.id, payload, file, reason)
  File "/home/fightman/.local/lib/python3.8/site-packages/discord/http.py", line 338, in request
    raise HTTPException(response, data)
discord.errors.HTTPException: 400 Bad Request (error code: 50046): Invalid Asset
```

Using `await ctx.send(file=file)` will send the actual image. 
Tried without `filename=` but nothing changed.
In the documents it clearly says `File`
image

### Intents

discord.Intents.all()

### System Information

- Python v3.8.10-final
- discord.py v2.0.0-alpha
    - discord.py pkg_resources: v2.0.0a3440+g3a451c9c
- aiohttp v3.6.2
- system info: Linux 5.4.0-80-generic #90-Ubuntu SMP Fri Jul 9 22:49:44 UTC 2021

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

Printing `self.client.stickers` will return the correct list of GuildSticker objects.

### PR DESCRIPTION
**Pull request by SebbyLaw**
## Summary

Resolves #7311 by making the `description` parameter of `Guild.create_sticker` required instead of Optional.
The documentation at https://discord.com/developers/docs/resources/sticker#create-guild-sticker marks description as non-optional

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/guild.py [hunk 1]
```diff
@@ -2402,7 +2402,7 @@ async def create_sticker(
         -----------
         name: :class:`str`
             The sticker name. Must be at least 2 characters.
-        description: Optional[:class:`str`]
+        description: :class:`str`
             The sticker's description. Can be ``None``.
```
Comment (Gobot1234) [inline discord/guild.py:2406]:
```suggestion
            The sticker's description.
```

File: discord/guild.py [hunk 2]
```diff
@@ -2427,8 +2427,7 @@ async def create_sticker(
             'name': name,
         }
```
Comment (Gobot1234) [inline discord/guild.py:2428]:
```suggestion
            'description': description,
        }
```

**Comment by dolfies** [comment]
Wouldn't this be better if it was kept as optional but just did `description or ''`.

Technically empty descriptions are supported like this and this prevents a breaking change.

**Comment by Rapptz** [comment]
This method was added in 2.0 so the breaking change doesn't have to be documented.

**Comment by dolfies** [comment]
> This method was added in 2.0 so the breaking change doesn't have to be documented.

Yeah but it'd be such an easy QoL change.

**Pull request merged**